### ISSUE TITLE
[2021] On the Stability of Fine-tuning BERT: Misconceptions, Explanations, and Strong Baselines

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#134]
## 0. 論文

| タイトル | On the Stability of Fine-tuning BERT: Misconceptions, Explanations, and Strong Baselines |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2006.04884 https://github.com/uds-lsv/bert-stable-fine-tuning |
| 著者    | Marius Mosbach, Maksym Andriushchenko, Dietrich Klakow |
| 投稿日付 | 2020/6/8 on arxiv (v1) |

## 1. どんなもの？

BERT をはじめとする事前学習言語モデルの fine-tuning における性能の不安定性に関して、その原因の分析を行い安定化の方法を提案した。一般に言及される原因である catastrophic forgetting とデータセットサイズの小ささは主要因ではないとし、分析の結果原因を optimization によるものと generalization によるものと判断し、それぞれで解決策を提案。
複数 seed で GLUE データセットに対して実験を行い、baseline や既存手法 (Mixout) に対して高い安定性と性能を確認した。
![スクリーンショット 2021-06-21 1 54 56](https://user-images.githubusercontent.com/1255485/122682332-b4000f80-d233-11eb-8f20-55886fc0214c.png)

## 2. 先行研究と比べてどこがすごい？

fine-tuning の不安定性の原因を分析し optimization と generalization に分解、それぞれで解決策を提案し実際に改善した。

## 3. 技術や手法のキモはどこ？

- 2つのよく述べられている要因について分析
    - catastrophic forgetting
        - 多少は影響はあるがこれが要因とは言えない
        - fine-tuning の失敗例と成功例それぞれで元の MLM タスクの精度を見たところ、どちらも catastropihc forgetting は発生している
    - データサイズの小ささ
        - 精度に影響はあるが不安定化の主要因とは言えない
        - 1k サンプルで 3 epoch と、もっと多くの epoch で実験したところ、より多くの training iteration により安定化
- 問題を optimization と generalization に分解し整理
    - optimization
        - BERTAdam で提案されている bias correction を除いたのが不安定化の原因
        - 元に戻す (bias correction を入れる) ことで改善
        - warm up と同等の効果だということで抜かれていたが、両方あることで安定化することが分かった
    - generalization
        - データサイズが小さくても training iteration が多ければ安定化する
        - train loss がほぼ 0 になっても generalization には影響なさそう
        - [IMO] 何が言いたいかよくわからん
- 最終的に次のベースラインを提案
    - 小さめの lr で optimization は bias correction を入れる
    - 学習はたっぷり 20 epoch (20 は例) 行い、training loss はしっかり下げる

## 4. どうやって有効だと検証した？

ベースライン (オリジナルBERT) と Mixout と比較し、GLUE データセットを対象に複数 seed で多くの実験を行い、実際にモデル精度を安定化することを確認した。
![スクリーンショット 2021-06-21 2 09 22](https://user-images.githubusercontent.com/1255485/122682655-c5e2b200-d235-11eb-9dd5-3feff7b6102c.png)

## 6. 次に読むべき論文は？

- Revisiting Few-sample BERT Fine-tuning
    - https://github.com/cfiken/paper-reading/issues/133
- Fine-Tuning Pretrained Language Models: Weight Initializations, Data Orders, and Early Stopping
    - https://github.com/cfiken/paper-reading/issues/43
